docs/move_clusters.md

```diff
@@ -0,0 +1,59 @@
+# Moving Hive-managed Clusters
+
+## Summary
+
+There is a need to be able to move Hive-managed clusters from one Hive instance to another.
+This may be useful if the user want to replace a Hive instance with a new one or if the user
+wants to implement sharding between Hive instances.
+
+Hive will manage all of the moving for the user. The user will create a CR indicating with
+details about the destination cluster and label selectors to match against the cluster to
+move.
+
+## ClusterRelocator CRD
+
+A ClusterRelocator is cluster-scoped.
```

Comment:
KB asked me privately if Hive could grow a feature someday to have multiple Hives installed in a single cluster, i.e. one on hot standby. I guess this would be cheaper than having an entire second Hive cluster on standby. IMO you lose fault tolerance if the cluster dies -- I'd rather just have a hot standby cluster. Thought I would mention it in case this is easy to accomplish without adding a cluster-scoped resource.

Reply:
> I'm not sure how this would help. What use cases are we trying to address with this?
> Hive relies on kube for HA.

That was pretty much my thought.

Reply:
Yeah on the surface I don't think this makes sense either, just bump your replicas and use leader election, there's nothing having another set of pods running would buy you. 

Separate hot backup hive clusters also sounds troubling, we don't want to be writing our own replication and failover code.